src/clincoded/static/components/variant_central/interpretation/functional.js

```diff
@@ -53,11 +62,97 @@ var CurationInterpretationFunctional = module.exports.CurationInterpretationFunc
                 setTimeout(scrollElementIntoView(evaluation_section_mapping[this.state.selectedCriteria], 'class'), 200);
             });
         }
+        if (this.props.functionalData && nextProps.interpretation && nextProps.interpretation.evaluations) {
+            this.compareFunctionalData(this.props.functionalData, nextProps.interpretation.evaluations);
+        }
+    },
+
+    compareFunctionalData: function(newFuncData, savedEvals) {
+        // Returns true if @newData is different from @oldData based on the unique @rev, otherwise returns false
+        const isDiffAfisVersion = (newAfis, oldAfis) => {
+            if (_.isEmpty(oldAfis) || _.isEmpty(newAfis)) {
+                return true;
+            }
+            const afisKeys = newAfis && Object.keys(newAfis);
+            return afisKeys.some(key => {
+                const oldStatements = oldAfis[key] && oldAfis[key].statements;
+                const newStatements = newAfis[key] && newAfis[key].statements;
+                if (!oldStatements || !newStatements || oldStatements.length !== newStatements.length) {
+                    return true;
+                }
+                return newStatements.some((statement, index) => {
+                    if (statement.rev !== oldStatements[index].rev) {
+                        return true;
+                    }
+                });
+            });
+        };
+        savedEvals.some(evaluation => {
+            if (['BS3', 'PS3'].indexOf(evaluation.criteria) > -1) {
+                const funcDataObjDiffFlag = isDiffAfisVersion(newFuncData, evaluation.functional.functionalData);
```

Comment:
```suggestion
                const funcDataObjDiffFlag = isDiffAfisVersion(newFuncData, evaluation.functional && evaluation.functional.functionalData);
```
Not sure that ```evaluation.functional``` is certain to exist...